Add sassDoc to component

JakeChampion opened the pull request:
Resolves #2 

# sass files to document
- [x] main.scss
- [x] [/src/scss/_variables.scss](https://github.com/Financial-Times/o-topper/blob/57a70a6dbc3d071b323c02e7f427f7c466ab4c93/src/scss/_variables.scss#L2)
- [ ] [/src/scss/_mixins.scss](https://github.com/Financial-Times/o-topper/blob/57a70a6dbc3d071b323c02e7f427f7c466ab4c93/src/scss/_mixins.scss#L1)
- [ ] [/src/scss/_layout.scss](https://github.com/Financial-Times/o-topper/blob/57a70a6dbc3d071b323c02e7f427f7c466ab4c93/src/scss/_layout.scss#L1)
- [x] [/src/scss/_grid.scss](https://github.com/Financial-Times/o-topper/blob/57a70a6dbc3d071b323c02e7f427f7c466ab4c93/src/scss/_grid.scss#L1)
- [ ] [/src/scss/_elements.scss](https://github.com/Financial-Times/o-topper/blob/57a70a6dbc3d071b323c02e7f427f7c466ab4c93/src/scss/_elements.scss#L1)
- [ ] [/src/scss/themes/_split-text.scss](https://github.com/Financial-Times/o-topper/blob/57a70a6dbc3d071b323c02e7f427f7c466ab4c93/src/scss/themes/_split-text.scss#L1)
- [x] [/src/scss/themes/_right-rail.scss](https://github.com/Financial-Times/o-topper/blob/57a70a6dbc3d071b323c02e7f427f7c466ab4c93/src/scss/themes/_right-rail.scss#L1)
- [ ] [/src/scss/themes/_package.scss](https://github.com/Financial-Times/o-topper/blob/57a70a6dbc3d071b323c02e7f427f7c466ab4c93/src/scss/themes/_package.scss#L1)
- [ ] [/src/scss/themes/_news-package.scss](https://github.com/Financial-Times/o-topper/blob/57a70a6dbc3d071b323c02e7f427f7c466ab4c93/src/scss/themes/_news-package.scss#L1)
- [x] [/src/scss/themes/_full-bleed-offset.scss](https://github.com/Financial-Times/o-topper/blob/57a70a6dbc3d071b323c02e7f427f7c466ab4c93/src/scss/themes/_full-bleed-offset.scss#L1)
- [x] [/src/scss/themes/_full-bleed-image.scss](https://github.com/Financial-Times/o-topper/blob/57a70a6dbc3d071b323c02e7f427f7c466ab4c93/src/scss/themes/_full-bleed-image.scss#L1)
- [x] [/src/scss/themes/_centered.scss](https://github.com/Financial-Times/o-topper/blob/57a70a6dbc3d071b323c02e7f427f7c466ab4c93/src/scss/themes/_centered.scss#L1)
- [x] [/src/scss/themes/_branded.scss](https://github.com/Financial-Times/o-topper/blob/57a70a6dbc3d071b323c02e7f427f7c466ab4c93/src/scss/themes/_branded.scss#L1)

JakeChampion committed:
Add sassDoc for main mixin for o-topper

origamiserviceuser commented:
o-topper bundle size difference from 3.1.0 to de46f9751ac23bc0b24a3e0cfd64cb143656b73a
No significant bundle size differences found.

JakeChampion committed:
Mark all private mixins as private via sassdoc

origamiserviceuser commented:
o-topper bundle size difference from 3.1.0 to 043998f39dbf745e2cfdff840867159814432f05
No significant bundle size differences found.

notlee committed:
Handle placeholder output the same as other components

Which we can remove with a move to Dart Dass. Document.

notlee committed:
Remove unneeded stylelint disable comment

notlee committed:
Add remaining `@access private` SassDoc

It is not really needed, an underscore is treated as private by SassDoc

notlee committed:
Remove unused private mixin.

notlee committed:
Organise and document the right-rail modifier.

- It's not really what a teaser considers a theme. It's a modifier
for multiple themes.
- It's app specific it seems.
- Private mixins are used in the app. That's for another commit to
deal with.

notlee committed:
Document src/scss/themes/_branded.scss

notlee committed:
Correct linting issues

origamiserviceuser commented:
o-topper bundle size difference from 3.1.1 to fa527bd81578815c6c3e6b6435c075a12909dc12
No significant bundle size differences found.

notlee committed:
Document the Full Bleed Image topper theme(s)

origamiserviceuser commented:
o-topper bundle size difference from 3.1.1 to f25795198b6396d4e56d172ec07a9d0c0ed6852c
css, master: 0.36kb decrease (-0.01kb/gzip)
An insignificant difference was also found for: js

notlee committed:
Document the Full Bleed Offset topper

origamiserviceuser commented:
o-topper bundle size difference from 3.1.1 to b69cbc8094575d27624b18189d2cd287ada3466d
css, master: 0.36kb decrease (-0.01kb/gzip)
An insignificant difference was also found for: js

notlee committed:
wip: add some todo notes

origamiserviceuser commented:
o-topper bundle size difference from 3.1.1 to e46a613b554c80163cca17961331e23418071acd
css, master: 0.36kb decrease (-0.01kb/gzip)
An insignificant difference was also found for: js

notlee committed:
Add missing stylelint dependency

This will allow IDEs to use the stylelint rules

notlee committed:
Add to package theme documentation

origamiserviceuser commented:
o-topper bundle size difference from 3.1.1 to 87f3d6d52abc57bbf6ce8b9378480ea70ce9f41e
css, master: 0.36kb decrease (-0.00kb/gzip)
An insignificant difference was also found for: js

JakeChampion committed:
Merge branch 'master' into sassdoc

JakeChampion committed:
Improve the focus outline contrast on topper links to meet contrast requirement
